### PR TITLE
Updated Docker Installation Alias for linux

### PR DESCRIPTION
**Pull request by Akash-Singh04**
## Related Issue
  - Only add sudo to docker alias for linux if required

Closes: #1469 

#### Describe the changes you've made
- We check if ```docker ps``` runs successfully without sudo.
- If it does, the alias is created without sudo, otherwise, it's created with sudo, otherwise not
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added
NIL
#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NIL
## Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by Akash-Singh04** [comment]
@slayerjain Kindly look into this PR for the issue you opened

**Comment by slayerjain** [comment]
@Akash-Singh04 thanks for the PR. 

@PranshuSrivastava since you were working on this script, can you please and test this?

**Comment by PranshuSrivastava** [comment]
Hey @Akash-Singh04, thanks for the PR. But this is not a good solution for the problem. This is kind of a hack to check if docker runs with sudo or not. So in macos sudo is generally not required, but in linux sudo is required but it can be removed with a specific configuration. So take a look at how the need for sudo is removed in Linux, and check if the user's system has that configuration or not, and if it does then don't use sudo

**Comment by Akash-Singh04** [comment]
In this updated implementation, we check if the current user belongs to the ```docker``` group using the ```groups``` command since users inside the ```docker``` group have permission to run docker without the ```sudo``` command

Reference: https://askubuntu.com/a/477554

File: keploy.sh [hunk 1]
```diff
@@ -71,20 +71,25 @@ installKeploy (){
...
-            if ! docker volume inspect debugfs &>/dev/null; then
-                docker volume create --driver local --opt type=debugfs --opt device=debugfs debugfs
-            fi
-            alias keploy='sudo docker run --pull always --name keploy-v2 -p 16789:16789 --privileged --pid=host -it -v $(pwd):$(pwd) -w $(pwd) -v /sys/fs/cgroup:/sys/fs/cgroup -v debugfs:/sys/kernel/debug:rw -v /sys/fs/bpf:/sys/fs/bpf -v /var/run/docker.sock:/var/run/docker.sock -v '"$HOME"'/.keploy-config:/root/.keploy-config -v '"$HOME"'/.keploy:/root/.keploy --rm ghcr.io/keploy/keploy'
+    if [ "$OS_NAME" = "Darwin" ]; then
+        if ! docker volume inspect debugfs &>/dev/null; then
+            docker volume create --driver local --opt type=debugfs --opt device=debugfs debugfs
+        fi
+        alias keploy='docker run --pull always --name keploy-v2 -p 16789:16789 --privileged --pid=host -it -v $(pwd):$(pwd) -w $(pwd) -v /sys/fs/cgroup:/sys/fs/cgroup -v debugfs:/sys/kernel/debug:rw -v /sys/fs/bpf:/sys/fs/bpf -v /var/run/docker.sock:/var/run/docker.sock -v '"$HOME"'/.keploy-config:/root/.keploy-config -v '"$HOME"'/.keploy:/root/.keploy --rm ghcr.io/keploy/keploy'
+    else
```
Comment (PranshuSrivastava) [inline keploy.sh:83]: use else if here, we are doing this specifically on linux

**Review by PranshuSrivastava — CHANGES_REQUESTED** [review]
The rest of the changes look good, there is just one comment that you need to address.

File: keploy.sh (hunk 1 above)
Comment (Akash-Singh04) [reply keploy.sh:83]: Kindly check if this was the intended change

**Review by PranshuSrivastava — APPROVED** [review]
LGTM

**Pull request merged**